### ISSUE TITLE
feature request—ability to add exceptions to b:closetag_emptyTags

### ISSUE THREAD

**Issue by jackkinsella** [alvan/vim-closetag#23]
This feature is motivated by the increasing use of JSX and Javscript Flow within the React.js world.

In general the auto-closing of tags by this plugin is a godsend for JSX users, but when the pluging interacts with Flow, Facebook's type-checking layer, it becomes too eager.

For example, sometimes I want to declare things like the following:

```
Promise<any>
React.Element<*>
```

As it stands, these two bad boys auto-expand (that rhymes lol!)

**Comment by alvan** [comment]
Hi @jackkinsella 
Do you have any suggestion for this feature?
Currently you can use `:CloseTagDisableBuffer` to temporarily disable the completion function.

**Comment by jackkinsella** [comment]
I guess the approach would be to define a `g:closetag_emptyTags` setting that can be used within .vimrc, the contents of which then gets appended to the internal `b:closetag_emptyTags`. 

I've never done any vim scripting before, so I wouldn't be able to do this without diving in pretty deep. But if I get a spare Saturday I might give it a shot.

**Comment by alvan** [comment]
Hi @jackkinsella 
The problem is, per your requirement, the `b:closetag_emptyTags` depends on the filetype you are editing like jsx, so it won't be a good idea that just add a global variable to define it.
And also, I am not sure if you can define a full list of all empty tags for jsx or not.
Because from your demo code, I saw you are typing `<any>` tag.

**Comment by jackkinsella** [comment]
I realised that my first message was a bit vague so I'll just clarify for anyone else reading the thread: I meant that in JSX/flow you sometimes literally type `<any>` and `<*>` i.e. both "any" and "*" are the actual tag names that need to be input, as opposed to placeholders/regexes for patterns of tags. 

But, clarification aside, you are right @alvan, there are other tags necessary and adding them all one-by-one could be tedious. 

Perhaps another, more general solution would be some keyword shortcut that closes the first tag without adding a second parallel closing tag. For example, imagine I have typed
`<any|` where | represents my cursor position. It would be cool if I could type some shortcut (e.g.  `<c->>` if possible ) to end up with `<any>`.
